### PR TITLE
pr request sent, ticket 7 almost done

### PR DESCRIPTION
**Pull request by Controllah-Gabi**
first review

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -161,4 +161,41 @@ describe("/api/articles/:article_id/comments",()=>{
...
+            }
+            return request(app)
+            .post("/api/articles/1/comments")
+            .send(commentData)
+            .expect(404)
+            .then((data)=>{
+                expect(data.body.msg).toBe("Username does not exist")
+            })
+        })
     })
```
Comment (emilyb93) [inline __tests__/app.test.js]: I would remove the article_id from the object here, you are already providing it in the url and as such you can take it from the url in your controller from req.params

Comment (emilyb93) [reply __tests__/app.test.js]: I think theres a few more tests here to do as well;
Posting to an article number that doesnt exist
posting to an article with an id that isnt even a number
posting a comment that doesnt even have a username
posting a comment that has a valid username but not a body?

File: utils/db.js [hunk 2]
```diff
@@ -18,3 +18,15 @@ exports.checkArticleExists = (article_id) => {
...
+        return pool.query(
+            'SELECT * FROM articles WHERE author =$1',[username]).then((data)=>{
+                if(data.rows.length === 0){
+                    return Promise.reject({
+                        status: 404,
+                        msg: "Username does not exist"
+                    })
+                }
+        });
+        }
```
Comment (emilyb93) [inline utils/db.js:31]: You dont need this anymore since you are checking this by dealing with foreign key errors :)

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Few more tests to do, but this is looking great so far!

**Review by hevdevs — CHANGES_REQUESTED** [review]
Testing is looks brill based on suggestions! I would have another look at what Emily is talking about with regards to username a bit more though, an invalid username value should be handled in your PSQL error handling middleware in your app (error code to look out for is 23503)

**Review by Harrison365 — CHANGES_REQUESTED** [review]
Hey Controllah :) ,

Please...
-Change the author in the req.body to username, I believe that's what the ticket asked for.
-Please test and error handle for article doesn't exist, invalid id (e.g. a string), req.body has missing fields (username, body, and both).

Thanks.

**Review by Harrison365 — APPROVED** [review]
Looks good :)

**Pull request merged**